From 4d3513e5a711ae0a69dfd08d0bde2f1e0fd1075d Mon Sep 17 00:00:00 2001
From: Noam Lessner <nlessner@habana.ai>
Date: Sun, 09 Feb 2020 08:18:37 +0200
Subject: [PATCH] [SW-8229] goya2: set flash access offset to 12MB

Flash region assigned to ARC is above 12MB. Adjust offset
registers to read images starting at 12MB.

Change-Id: I7d4bf8b2caf69e7757e61482885b14e90d9cafff
Signed-off-by: Noam Lessner <nlessner@habana.ai>
---

diff --git a/common/spi_flash.h b/common/spi_flash.h
index 3dcea5b..3a75f35 100644
--- a/common/spi_flash.h
+++ b/common/spi_flash.h
@@ -40,12 +40,13 @@
 #define FLASH_OTP_FW_VERSION_OFFSET	160
 #define FLASH_REVOKE_BITS_PER_DWORD	4
 
-#define FLASH_PRIMARY_OFFSET		0
+#define FLASH_ARC_BASE_ADDRESS		0xC00000
+#define FLASH_PRIMARY_OFFSET		FLASH_ARC_BASE_ADDRESS
 #define FLASH_SECONDARY_OFFSET		(FLASH_PRIMARY_OFFSET + \
 						sizeof(uint32_t))
 #define FLASH_EFUSE_OFFSET		(FLASH_SECONDARY_OFFSET + \
 						sizeof(uint32_t))
-#define FLASH_IMAGE_SWITCH_OFFSET	4096
+#define FLASH_IMAGE_SWITCH_OFFSET	(FLASH_ARC_BASE_ADDRESS + 4096)
 
 /* 16 byte */
 #define FLASH_EFUSE_FW_VERSION_OFFSET	0
